### ISSUE TITLE
btrbk fails during incremental send/receive with ERROR referring to "setxattr"

### ISSUE THREAD

**Issue by 2mc** [digint/btrbk#292]
Hello, 
I enjoy btrbk for years! 

But now I hit on an unusual ERROR message. Here is the context: 

```
Archiving subvolumes: /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.*
Creating incremental backup...
[send/receive] target: /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190908T0645
[send/receive] source: /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190908T0645
[send/receive] parent: /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190907T0502
[send/receive] checking target metadata: /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190908T0645
Creating incremental backup...
[send/receive] target: /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648
[send/receive] source: /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648
[send/receive] parent: /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190908T0645
WARNING: [send/receive] (send=/storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648, receive=/storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps) At subvol /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648
WARNING: [send/receive] (send=/storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648, receive=/storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps) ERROR: lsetxattr o257-128243-0 system.posix_acl_default= failed: No such file or directory
WARNING: [send/receive] (send=/storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648, receive=/storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps) At snapshot @VMs.20190909T0648
[send/receive] checking target metadata: /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648
ERROR: Failed to send/receive btrfs subvolume: /storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648 [/storage/T3ButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190908T0645] -> /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps
[delete] options: commit-after
[delete] target: /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648
WARNING: Deleted partially received (garbled) subvolume: /storage/4TButterBAK/@backups/_shared/btrbk_VMs_snaps/@VMs.20190909T0648
ERROR: Error while archiving subvolumes, aborting
Archived 1/5 subvolumes
```
I tried to archive the failing subvolume `@VMs.20190909T0648` onto another disk but got the same results.

Could that mean that the source subvolume is corrupted or might it indicate a btrbk or btrfs bug? If it is the former, how would I check its integrity. If it is the latter, how shall we proceed?

Thank you, indeed.

**Comment by digint** [comment]
> ERROR: lsetxattr o257-128243-0 system.posix_acl_default= failed: No such file or directory

It's related/duplicate of #91, #223, #225, #251 : upstream (btrfs) bug, nothing that can be fixed in btrbk.

Some times deleting the parent subvolume helps, some times deleting the source subvolume helps. I'm afraid I can't give you much more advice on this...

For the record, what kernel / btrfs-progs version do you have?

**Comment by 2mc** [comment]
Thank you for pointing me to these issues. 
It took a while to read through them.
Sad enough, I am obviously in that boat, too.

For the record:
It happened on archlinux, kernel 5.2.14-arch2-1-ARCH, btrfs-progs v5.2.1

**Comment by 2mc** [comment]
More to the point and further for the record:

I discovered a pattern in my send/receive troubles. They all happened during incremental archiving to another disk of a series of snapshots backed up before. The retention policy specified to keep one snapshot for all years and 12 month of the present year. It was always the oldest monthly snapshot that could not be send incrementally. That is, right at the retention policy boarder. Manually, I could send all the failing snapshots non-incremental (as full backup). After this manual intervention, I could continue with the incremental archive command of btrbk without problems. 

So, obviously incremental send failed because necessary extents could not be found. If this is really due to a bug in `btrfs send` or, if I broke the tree of parents some year(s) ago, I don't know, cannot recall nor reproduce.

However, when archiving the now archived backup disk once more, I will hit on the same time-consuming issue of manual interaction.

**Proposal:**
Maybe the btrbk archive command could provide an option that btrbk will try to re-send a snapshot that failed to be sent incrementally -- and then, if that worked out, to continue with incremental archiving?
